### PR TITLE
Add simulation API for testing existing rule

### PR DESCRIPTION
**Pull request by yibirnba**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json [hunk 1]
```diff
@@ -194,6 +188,90 @@
...
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/autoDismissAlertsRules/{autoDismissAlertsRuleName}/simulation": {
+      "get": {
+        "x-ms-examples": {
+          "Get auto dismiss alert rule simulation for subscription": {
+            "$ref": "./examples/AutoDismissAlertsRules/GetAutoDismissAlertsRuleSimulationList_example.json"
+          }
+        },
+        "tags": [
+          "AutoDismissAlertsRules"
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json]: Check if you need to update this

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json [hunk 2]
```diff
@@ -232,123 +325,39 @@
...
-          "type": "string",
-          "format": "date-time",
-          "description": "The last time this rule was modified"
-        },
-        "expirationDateUTC": {
+        "simulatedAlertId": {
           "type": "string",
-          "format": "date-time",
-          "description": "Expiration date of the rule"
+          "description": " "
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json]: Can be removed?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json [hunk 3]
```diff
@@ -232,123 +325,39 @@
...
-          "type": "object",
-          "description": "The auto dismiss conditions",
-          "$ref": "#/definitions/AutoDismissAlertsScope"
-        },
-        "ruleImpact": {
-          "type": "object",
-          "description": "The impact of the rule. Use for testing the rule and will be returned only when the test flag is set",
-          "readOnly": true,
-          "$ref": "#/definitions/RuleImpact"
+          "description": " "
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissAlertsRules.json]: Can be removed?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRuleSimulationList_example.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "200": {
+      "body": {
+        "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/locations/westeurope/AutoDismissAlertsRules/dismissIpAnomalyAlerts/simulation/2518770965529163669_F144EE95-A3E5-42DA-A279-967D115809AA",
+        "name": "2518770965529163669_F144EE95-A3E5-42DA-A279-967D115809AA",
+        "type": "Microsoft.Security/Locations/AutoDismissAlertsRules/Simulation",
+        "properties": {
+          "simulatedAlertId": "2518770965529163669_F144EE95-A3E5-42DA-A279-967D115809AA",
+          "simulatedState": "Dismissed"
+        }
+      }
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRuleSimulationList_example.json]: I think that this needs to return an array.

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRuleSimulation_example.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+              "simulatedState": "Dismissed"
+            }
+          },
+          {
+            "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/locations/westeurope/AutoDismissAlertsRules/dismissIpAnomalyAlerts/simulation/2518765996949954086_2325cf9e-42a2-4f72-ae7f-9b863cba2d22",
+            "name": "2518765996949954086_2325cf9e-42a2-4f72-ae7f-9b863cba2d22",
+            "type": "Microsoft.Security/Locations/AutoDismissAlertsRules/Simulation",
+            "properties": {
+              "simulatedAlertId": "2518765996949954086_2325cf9e-42a2-4f72-ae7f-9b863cba2d22",
+              "simulatedState": "Active"
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRuleSimulation_example.json]: See the name of the file it needs to return a single element

**Review by nitsi — APPROVED** [review]

**Comment by orparnes** [comment]
>   "swagger": "2.0",

Please don't document it in this preview version, but only on the stable version.

You can still support it in the preview controller, but as it is a new feature I prefer we'll be committed to it only on newer versions, so deprecation of old versions will be easier in the future

It is also relevant to the sample files

---
Refers to: specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/alerts.json:2 in 9fa92fb. [](commit_id = 9fa92fb98a01ef6a0916bdf2554a7c5084b16cc4, deletion_comment = False)

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+              "state": "Enabled",
+              "reason": "Detects a normal activity (False Positive)",
+              "comment": "Test VM",
+              "autoDismissAlertsScope": {
+                "allOf": [
+                  {
+                    "field": "CompromisedEntity",
+                    "in": [
+                      "ContosoVM1",
+                      "demoAccount@microsoft.com"
```
Comment (orparnes) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json:28]: >microsoft [](start = 35, length = 9)

suggesting to use "contoso.com" as it is usually the fake company used by MS demos

**Comment by orparnes** [comment]
>     "200": {

200 or 204?

Relevant also for other examples

---
Refers to: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRules_example.json:7 in 9fa92fb. [](commit_id = 9fa92fb98a01ef6a0916bdf2554a7c5084b16cc4, deletion_comment = False)

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{
+  "parameters": {
+    "api-version": "2019-01-01-preview",
+    "subscriptionId": "20ff7fc3-e762-44dd-bd96-b71116dcdc23",
+    "alertType": "IpAnomaly"
+  },
+  "responses": {
+    "200": {
```
Comment (orparnes) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json:8]: >200 [](start = 5, length = 3)

200 or 204?

Relevant also for other examples

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Alerts/GetAlertsResourceGroupWithAutoDismissRule_example.json [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+              "vendorName": "Microsoft",
+              "alertDisplayName": "Threat Intelligence Alert",
+              "alertName": "ThreatIntelligence",
+              "detectedTimeUtc": "2018-05-01T19:50:47.083633Z",
+              "description": "Process was detected running on the host and is considered to be suspicious, verify that the user run it",
+              "remediationSteps": "verify that the user invoked this process\r\nrun antimalware scan of the VM",
+              "actionTaken": "Detected",
+              "reportedSeverity": "High",
+              "compromisedEntity": "vm1",
+              "associatedResource": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/myRg1/providers/Microsoft.Compute/virtualMachines/vm1",
```
Comment (orparnes) [inline specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Alerts/GetAlertsResourceGroupWithAutoDismissRule_example.json:26]: >20ff7fc3-e762-44dd-bd96-b71116dcdc23 [](start = 52, length = 36)

please verify that the subscription id in the examples is aligned with other examples of Microsoft.Security APIs

(I'm not sure if it is actually the subscription used to simulate alerts and generate the SDK libraries or it is a fake guid..)

**Comment by orparnes** [comment]
>   "swagger": "2.0",

looks good

---
Refers to: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/alerts.json:2 in 9fa92fb. [](commit_id = 9fa92fb98a01ef6a0916bdf2554a7c5084b16cc4, deletion_comment = False)

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json (hunk 7 above)
Comment (orparnes) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json:8]: (I just saw some place uses 204. Not sure what each means and if there is any place with wrong value...)

---
In reply to: [351815292](https://github.com/yibirnba/azure-rest-api-specs/pull/5#discussion_r351815292) [](ancestors = 351815292)

File: specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Alerts/GetAlertsResourceGroupWithAutoDismissRule_example.json (hunk 8 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/examples/Alerts/GetAlertsResourceGroupWithAutoDismissRule_example.json:26]: yes it is in most API examples

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json (hunk 7 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json:8]: 204 it is ok with no content. here we return content

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json (hunk 6 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissAlertsRules/GetAutoDismissAlertsRulesWithAlertType_example.json:28]: done

**Pull request closed without merging**